src/dialects/sqlite/query-interface.js

```diff
@@ -23,11 +23,41 @@ class SQLiteQueryInterface extends QueryInterface {
     const fields = await this.describeTable(tableName, options);
     delete fields[attributeName];
 
-    const sql = this.queryGenerator.removeColumnQuery(tableName, fields);
-    const subQueries = sql.split(';').filter(q => q !== '');
+    const savepointName = 'remove_column_savepoint';
 
-    for (const subQuery of subQueries) {
-      await this.sequelize.query(`${subQuery};`, { raw: true, ...options });
+    // Disable foreign key constraints for the next sequence of operations
+    await this.sequelize.query('PRAGMA foreign_keys = OFF;', options);
+    // Use SAVEPOINT to allow for nested transactions
+    await this.sequelize.query(`SAVEPOINT ${savepointName};`, options);
```

Comment:
if this fails for any reason (such as network issue). `PRAGMA foreign_keys` will not be reverted.

There are a few places where `PRAGMA foreign_keys` is used. It would be better extracted to a helper function:

```typescript
async withForeignKeysOff(options, cb) {
  await this.sequelize.query('PRAGMA foreign_keys = OFF', options);
  
  try {
    return await cb();
  } finally {
    await this.sequelize.query('PRAGMA foreign_keys = ON', options);
  }
}

// ...

this.withForeignKeysOff(options, () => {
  await this.sequelize.query(`SAVEPOINT ${savepointName};`, options);
  // ...
});
```

Reply:
Looks good, I will redo the logic inside ```changeColumn``` and ```removeColum``` using the above abstraction.